src/train.jl

```diff
@@ -55,9 +55,11 @@ function train(hybridModel, data, save_ps; nepochs=200, batchsize=10, opt=Adam(0
             # ? check NaN indices before going forward, and pass filtered `x, y`.
             is_no_nan = .!isnan.(y)
             if length(is_no_nan)>0 # ! be careful here, multivariate needs fine tuning
-                grads = Zygote.gradient((ps) -> lossfn(hybridModel, x, (y, is_no_nan), ps, st,
-                    LoggingLoss(training_loss=training_loss, agg=agg)), ps)[1]
+                l, backtrace = Zygote.pullback((ps) -> lossfn(hybridModel, x, (y, is_no_nan), ps, st,
+                    LoggingLoss(training_loss=training_loss, agg=agg)), ps)
+                grads = backtrace(l)[1]
                 Optimisers.update!(opt_state, ps, grads)
+                st = l[2].st
```

Comment:
wow! this looks so inconvenient. But if it is the way it is, nothing to do here, other than use it 😄 .